R/motifs.R

```diff
@@ -215,22 +229,30 @@ count_motifs <- function(graph, size = 3, cut.prob = rep(0, size)) {
 #' motifs(g, 3)
 #' count_motifs(g, 3)
 #' sample_motifs(g, 3)
-sample_motifs <- function(graph, size = 3, cut.prob = rep(0, size),
+sample_motifs <- function(graph, size = 3, cut.prob = NULL,
                           sample.size = vcount(graph) / 10, sample = NULL) {
   ensure_igraph(graph)
+  if (!is.null(cut.prob) && length(cut.prob) != size) {
   cut.prob <- as.numeric(cut.prob)
-  if (length(cut.prob) != size) {
     cut.prob <- c(
       cut.prob[-length(cut.prob)],
       rep(cut.prob[-length(cut.prob)], length(cut.prob) - 1)
     )
   }
 
   on.exit(.Call(R_igraph_finalizer))
-  .Call(
-    R_igraph_motifs_randesu_estimate, graph, as.numeric(size),
-    as.numeric(cut.prob), as.numeric(sample.size), as.numeric(sample)
-  )
+  if (is.null(cut.prob)) {
+    .Call(
+      R_igraph_motifs_randesu_estimate, graph, as.numeric(size),
+      cut.prob, as.numeric(sample.size), as.numeric(sample)
+    )
+  } else {
+    .Call(
+      R_igraph_motifs_randesu_estimate, graph, as.numeric(size),
+      as.numeric(cut.prob), as.numeric(sample.size), as.numeric(sample)
+    )
+
+  }
```

Comment:
Instead of putting the call in a conditional, update the value of `cut.prob` in a conditional, and keep a single call to the C function.